osbrain/tests/test_agent_async_requests.py

```diff
@@ -75,17 +76,13 @@ def test_return(nsproxy, server_client_late_reply_return):
     assert time.time() - t0 < 0.1
 
     # Server should receive first request "soon"
-    time.sleep(0.1)
-    assert server.get_attr('received') == ['foo']
-    assert client.get_attr('received') == []
+    assert wait_agent_attr(server, data='foo', timeout=0.1)
+    assert wait_agent_attr(client, value=[], timeout=0.1)
 
     # Wait for client to receive reply asynchronously
-    time.sleep(1)
-    assert server.get_attr('received') == ['foo', 'bar']
-    assert client.get_attr('received') == ['xfoo']
-    time.sleep(1)
-    assert server.get_attr('received') == ['foo', 'bar']
-    assert client.get_attr('received') == ['xfoo', 'xbar']
+    assert wait_agent_attr(client, data='xfoo')
+    assert wait_agent_attr(server, data='bar')
+    assert wait_agent_attr(client, data='xbar')
```

Comment:
Instead of `data=...` use `value=...`. The attribute is well known and not too long. Also, order in which those messages are received matters, so `data=...` is not as reliable.

The same applies for all `data=...` in this PR.

Reply:
What's the difference, then? Both `data` and `value` are correct in my opinion, but I figured `data` was a bit more clear, since it only includes the new value we are expecting to find, rather than the entire list with previously known values.

But now that you mention the order thing, why not change `data` to `endswith`? Would that be acceptable to you?

Reply:
Yeah, that should be fine.